dvc/output.py

```diff
@@ -965,7 +965,7 @@ def relative_update(self, inc: int = 1) -> None:
         return added, False if added else modified
 
     def remove(self, ignore_remove=False):
-        self.fs.remove(self.fs_path, recursive=True)
+        self.fs.remove(self.fs_path, recursive=False)
```

Comment:
We've discussed this before, but it loks wrong. E.g. imagine a directory with files as an output, this will error-out (at least unless I'm missing something). Could you elaborate on the problem that you are trying to solve, please?

Reply:
That's what the test tries to cover unless I'm misunderstanding you, but we probably do need to test it across other filesystems. 

The problem is in #8757 - the recursive call errors out for a nonexistent path in s3fs. 

As mentioned above, I don't mind closing if we come up with a different approach to solve it in s3fs or fsspec, but wanted to open it for discussion as a potential quick fix.

Reply:
I think you are spot on in your explanation in the issue. localfs is ignoring that flag (looks like our oversight) and since all other filesystems would also raise FileNotFoundError, we should just catch and ignore FileNotFoundError here as well.

```suggestion
        try:
            self.fs.remove(self.fs_path, recursive=True)
        except FileNotFoundError:
            pass
```

The test you've added could be removed. Ideally we would test external outputs in `dvc.testing`, but that will probably just have to wait for better times.

Reply:
The test will become relevant once we make our localfs comply with the flag, but IIRC there are a few more places that need to be fixed along the way, so that could wait.